src/integrationTest/java/com/mongodb/hibernate/query/select/SimpleSelectQueryIntegrationTests.java

```diff
@@ -0,0 +1,186 @@
+
+    @BeforeEach
+    void beforeEach(SessionFactoryScope scope) {
+        scope.inTransaction(session -> List.of(
+                        new Contact(1, "Bob", 18, Country.USA),
+                        new Contact(2, "Mary", 35, Country.CANADA),
+                        new Contact(3, "Dylan", 7, Country.CANADA),
+                        new Contact(4, "Lucy", 78, Country.CANADA),
+                        new Contact(5, "John", 25, Country.USA))
+                .forEach(session::persist));
+    }
+
+    @Test
+    void testComparisonByEQ(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country = :country",
+                q -> q.setParameter("country", Country.USA.name()),
+                List.of(1, 5)));
+    }
+
+    @Test
+    void testComparisonByNE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country != ?1",
+                q -> q.setParameter(1, Country.USA.name()),
+                List.of(2, 3, 4)));
+    }
+
+    @Test
+    void testComparisonByLT(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age < :age", q -> q.setParameter("age", 35), List.of(1, 3, 5)));
+    }
+
+    @Test
+    void testComparisonByLTE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age <= ?1", q -> q.setParameter(1, 35), List.of(1, 2, 3, 5)));
+    }
+
+    @Test
+    void testComparisonByGT(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age > :age", q -> q.setParameter("age", 18), List.of(2, 4, 5)));
+    }
+
+    @Test
+    void testComparisonByGTE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age >= :age", q -> q.setParameter("age", 18), List.of(1, 2, 4, 5)));
+    }
+
+    @Test
+    void andFilterTest(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country = ?1 and age > ?2",
+                q -> q.setParameter(1, Country.CANADA.name()).setParameter(2, 18),
+                List.of(2, 4)));
+    }
+
+    @Test
+    void orFilterTest(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country = :country or age > :age",
+                q -> q.setParameter("country", Country.CANADA.name()).setParameter("age", 18),
+                List.of(2, 3, 4, 5)));
+    }
+
+    @Test
+    void testFieldNonNumericLiteralValue(SessionFactoryScope scope) {
```

Comment:
There is more than one non-numeric literal value type, and as it stands this test is essentially the same as `testFieldStringLiteralValue` below.  

Let's test all the literal value types that are supported, each with its own test, e.g. `testFieldBooleanLiteralValue`

Reply:
I added a separate testing class for query literal testing, and removed the duplicated ones from `SimpleSelectQueryIntegrationTests`.